staging/src/k8s.io/client-go/util/cert/pem.go

```diff
@@ -150,7 +150,19 @@ func ParsePublicKeysPEM(keyData []byte) ([]interface{}, error) {
 // ParseCertsPEM returns the x509.Certificates contained in the given PEM-encoded byte array
 // Returns an error if a certificate could not be parsed, or if the data does not contain any certificates
 func ParseCertsPEM(pemCerts []byte) ([]*x509.Certificate, error) {
-	ok := false
+	certs, err := ParseEmptyCertsPEM(pemCerts)
+	if err != nil {
+		return certs, err
+	}
+	if len(certs) == 0 {
+		return certs, errors.New("data does not contain any valid RSA or ECDSA certificates")
+	}
+	return certs, nil
+}
+
+// ParseEmptyCertsPEM returns the x509.Certificates contained in the given PEM-encoded byte array
+// Returns an error if a certificate could not be parsed, or an empty slice if the data does not contain any certificates
+func ParseEmptyCertsPEM(pemCerts []byte) ([]*x509.Certificate, error) {
```

Comment:
I am struggling to understand the difference between this and ParseCertsPEM.
Is the difference that ParseEmptyCertsPEM allows empty pemCerts?
Consider adding an exported `ErrNoCerts` that can be checked by the caller instead.

Reply:
Good call, fixed